Add instructions to build on arm64 macOS

The current build instructions do not support building the application on macOS.  I tried using `BUILD_LINUX.md`, specifically:

```
NUM_CORES=14 &&
mkdir OpenLIFU-superbuild &&
cd OpenLIFU-superbuild &&
cmake -DCMAKE_BUILD_TYPE:STRING=Release -DQt5_DIR:PATH=/usr/local/opt/qt@5/lib/cmake/Qt5 -S ../OpenLIFU-app -B . &&
make -j${NUM_CORES}
```

And the configure step works, but at the build step, there are many linker errors.  Here is one example of the make call stack:

```
ld: symbol(s) not found for architecture arm64
c++: error: linker command failed with exit code 1 (use -v to see invocation)
make[5]: *** [bin/CTKAppLauncher] Error 1
make[4]: *** [CMakeFiles/CTKAppLauncher.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [slicersources-build/CTKAppLauncherLib-prefix/src/CTKAppLauncherLib-stamp/CTKAppLauncherLib-build] Error 2
make[1]: *** [slicersources-build/CMakeFiles/CTKAppLauncherLib.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Build issues on arm64 macOS are likely partially related to [Slicer#6811](https://github.com/Slicer/Slicer/issues/6811). I have addressed building Slicer in particular in a [Slicer discussion post](https://discourse.slicer.org/t/developing-for-slicer-on-apple-silicon-build-targeting-x86-64/40686).  The best we can get is a build targeting x86_64 and run through Rosetta 2, but this is how it is currently done by Slicer/Slicer.

--- a/BUILD_MACOS.md
+++ b/BUILD_MACOS.md
@@ -1,0 +1,67 @@
+# Build and Package OpenLIFU
+
+This document summarizes how to build and package OpenLIFU on macOS.
+
+OpenLIFU is a custom Slicer application. Reading the [3D Slicer Developer Documentation](https://slicer.readthedocs.io/en/latest/developer_guide/index.html) may help answer additional questions.
+
+The initial source files were created using [KitwareMedical/SlicerCustomAppTemplate](https://github.com/KitwareMedical/SlicerCustomAppTemplate).
+
+## Clone
+
+```sh
+git clone https://github.com/OpenwaterHealth/OpenLIFU-app.git
+```
+
+## Prerequisites
+
+1. Install the latest Xcode SDKs:
+    ```sh
+    xcode-select --install
+    ```
+
+2. Install Homebrew and Qt5:  
+    - **Apple Silicon:** Install `x86_64` Homebrew and Qt5. You can install "x86_64 Homebrew" if you run the `x86_64` shell with Rosetta 2. **Make sure you remain in the x86_64 shell for the rest of the build**:
+      ```sh
+      arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+      exec /usr/bin/arch -x86_64 /bin/bash --norc --noprofile
+      export PATH="/usr/local/bin:$PATH"
+      brew install qt5
+      ```
+    - **Intel:** Install Homebrew and Qt5:
+      ```sh
+      /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+      export PATH="/usr/local/bin:$PATH"
+      brew install qt5
+      ```    ```
+
+## Build
+
+Note: The build process can take hours.
+
+```sh
+mkdir OpenLIFU-superbuild
+cd OpenLIFU-superbuild
+cmake \
+-DQt5_DIR:PATH=/usr/local/opt/qt@5/lib/cmake/Qt5 \
+-DCMAKE_OSX_ARCHITECTURES:STRING=x86_64 \
+-DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=13.0 \
+-DCMAKE_BUILD_TYPE:STRING=Release \
+-DSlicer_USE_SYSTEM_QT:BOOL=ON \
+-DSlicer_USE_SimpleITK:BOOL=OFF \
+-B . \
+-S ../OpenLIFU-app
+
+NUM_CORES=$(sysctl -n hw.ncpu)
+make -j${NUM_CORES}
+```
+
+Once the application is built, there will be an "inner build" inside the superbuild folder, located at `OpenLIFU-superbuild/Slicer-build`. The application executable is contained in this inner build folder.
+
+## Package
+
+Enter the inner build folder and then make the package:
+
+```sh
+cd Slicer-build/
+make package -j${NUM_CORES}
+```

